processor/src/main/java/org/mapstruct/ap/internal/conversion/Conversions.java

```diff
@@ -352,15 +366,15 @@ public boolean equals(Object obj) {
                     return false;
                 }
             }
-            else if ( !sourceType.equals( other.sourceType ) ) {
+            else if ( !sourceType.erasure().equals( other.sourceType.erasure() ) ) {
```

Comment:
This is another point where I was not sure. If they are removed then the conversions are not found.
It seems that the type `Iterable<T>` is not the same to `Iterable<T>` (I don't know why)